framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -119,279 +82,156 @@ private static Pattern anchored(String regex) {
-                message += " in a context with parsingMember=true";
+                String message;
+                if (expr.toString().equals("_param_0")) {
+                    message =
+                            "one should use \"this\" for the receiver or \"#1\" for the first formal parameter";
+                } else {
+                    message = "is an unrecognized expression";
+                }
+                if (context.parsingMember) {
+                    message += " in a context with parsingMember=true";
+                }
+                throw constructParserException(expr.toString(), message);
             }
-            throw constructParserException(expression, message);
         }
-    }
 
-    private static boolean isMemberSelect(String s) {
-        return parseMemberSelect(s) != null;
+        return null;
     }
 
-    /**
-     * Matches a field access. First of returned pair is object and second is field.
-     *
-     * @param s expression string
-     * @return pair of object and field
-     */
-    private static Pair<String, String> parseMemberSelect(String s) {
-        Pair<Pair<String, String>, String> method = parseMethodCall(s);
-        if (method != null && method.second.startsWith(".")) {
-            return Pair.of(
-                    method.first.first + "(" + method.first.second + ")",
-                    method.second.substring(1));
-        }
-
-        Pair<Pair<String, String>, String> array = parseArray(s);
-        if (array != null && array.second.startsWith(".")) {
-            return Pair.of(
-                    array.first.first + "[" + array.first.second + "]", array.second.substring(1));
-        }
-
-        Matcher m = STRING_SELECT_PATTERN.matcher(s);
-        if (m.matches()) {
-            return Pair.of(m.group(1), m.group(2));
-        }
-
-        int nextRParenPos = matchingCloseParen(s, 0, '(', ')');
-        if (nextRParenPos != -1) {
-            if (nextRParenPos + 1 < s.length() && s.charAt(nextRParenPos + 1) == '.') {
-                String receiver = s.substring(0, nextRParenPos + 1);
-                String remaining = s.substring(nextRParenPos + 2);
-                return Pair.of(receiver, remaining);
-            } else {
-                return null;
-            }
-        }
+    /** Replaces every occurrence of "#(number)" with "_param_(number)" */
+    private static String noHashTags(String expression) {
+        String updatedExpression = expression;
 
-        int i = s.indexOf(".");
-        if (i == -1) {
-            return null;
+        for (Integer integer : parameterIndices(expression)) {
+            updatedExpression = updatedExpression.replaceAll("#" + integer, "_param_" + integer);
         }
 
-        String receiver = s.substring(0, i);
-        String remaining = s.substring(i + 1);
-
-        return Pair.of(receiver, remaining);
+        return updatedExpression;
     }
 
     private static Receiver parseMemberSelect(
```

Comment:
Please add a comment that the selected member could be a field, class, or package; but it can't be a field.  Also, many methods are missing Javadoc in this class; please make add documentation to every method.  If it's a method you didn't add and don't understand, let me know and I can document it.  Thanks!